### PR TITLE
Fix regular users were able to retrieve list of all machines

### PR DESCRIPTION
**Pull request by corentindrouet**
In response to issue #49 

replace machine() with getMachineForUser() for list only machines of the user
getMachineForUser don't return an array, add [].

### CODE REVIEW

File: api/controllers/MachineController.js [hunk 1]
```diff
@@ -33,9 +33,9 @@ module.exports = {
    * @method find
    */
   find(req, res) {
-    MachineService.machines()
+    MachineService.getMachineForUser(req.user)
```
Comment (dynamiccast) [inline api/controllers/MachineController.js]: Admins should still be able to list all machines

Comment (dynamiccast) [inline api/controllers/MachineController.js]: The same kind of filtering should be done on `connections` too.

File: api/controllers/MachineController.js [hunk 2]
```diff
@@ -51,10 +59,18 @@ module.exports = {
    * @method users
    */
   users(req, res) {
-    Machine.find()
-      .then((machines) => {
-        return res.ok(machines);
-      })
-      .catch((err) => res.negotiate(err));
+    if (req.user.isAdmin) {
+      Machine.find()
```
Comment (dynamiccast) [inline api/controllers/MachineController.js:63]: Actually, `/api/machines/users` MUST always return `getMachineForUser` regardless of user's status.

File: tests/api/machine.test.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ * along with this program.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/* globals sails, AccessToken, User, Machine */
+
+var nano = require('./lib/nanotest');
+
+module.exports = function() {
+  describe('machine', function() {
```
Comment (dynamiccast) [inline tests/api/machine.test.js:30]: This test should have an `after` hook to cleanup created users and machine.

File: tests/api/machine.test.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+          {
+            name: 'essai2',
+            type: 'manual',
+            ip: '2.2.2.2',
+            username: 'Administrator',
+            password: 'admin'
+          });
+        })
+        .then(() => {
+          return nano.request(sails.hooks.http.app)
```
Comment (dynamiccast) [inline tests/api/machine.test.js:57]: It is preferable to create user's token by directly calling `AccessToken.create`.
Here is an example: https://github.com/Nanocloud/nanocloud/blob/faa23db05f0931db340c63a975641b0e28bb39a0/tests/api/apps.test.js#L210

File: tests/api/machine.test.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+ */
+
+/* globals sails, AccessToken, User, Machine */
+
+var nano = require('./lib/nanotest');
+
+module.exports = function() {
+  describe('machine', function() {
+    it('Should return user machines', function(done) {
+      User.create({
```
Comment (dynamiccast) [inline tests/api/machine.test.js:32]: Users and apps (and more generally all pre requisites for a test should be create in a `before` hook.

File: tests/api/machine.test.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            .send({
+              username: 'test@test.com',
+              password: 'tests',
+              grant_type: 'password'
+            })
+            .set('Authorization', 'Basic ' + new Buffer('9405fb6b0e59d2997e3c777a22d8f0e617a9f5b36b6565c7579e5be6deb8f7ae:9050d67c2be0943f2c63507052ddedb3ae34a30e39bbbbdab241c93f8b5cf341').toString('base64'))
+            .expect(200);
+        })
+        .then(() => {
+          return User.find({
```
Comment (dynamiccast) [inline tests/api/machine.test.js:68]: Prefer `Model.findOne` if you expect only one result.

File: tests/api/machine.test.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+          });
+        })
+        .then((token) => {
+          nano.request(sails.hooks.http.app)
+            .get('/api/machines')
+            .set('Authorization', 'Bearer ' + token[0].token)
+            .expect(200)
+            .expect((res) => {
+              if (res.body.data.length > 1) {
+                throw new Error('Normal user should not be able to list all machines');
```
Comment (dynamiccast) [inline tests/api/machine.test.js:84]: Normal user -> Regular users

File: tests/api/machine.test.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+            });
+        })
+        .then(() => {
+          nano.request(sails.hooks.http.app)
+            .get('/api/machines/users')
+            .set(nano.adminLogin())
+            .expect(200)
+            .expect((res) => {
+              if (res.body.data.length <= 1) {
+                throw new Error('Admin user should be able to list all machines');
```
Comment (dynamiccast) [inline tests/api/machine.test.js:95]: Admin user -> Admins

**Comment by Leblantoine** [comment]
Closed in favor of PR : #91 
The comments have been considered.

**Pull request closed without merging**